### ISSUE TITLE
flet build apk does not work with ultralytics. ERROR:Could not find a version that satisfies the requirement polars... (and other packages)

### ISSUE THREAD

**Issue by DaniFernandesDesigner** [ultralytics/ultralytics#22204]
I am using Windows 10, Pycharm 21.3.2, with Python 3.10 and YOLO Ultralytics to design an app for Android. And I am using Flet framework for that. Everything was working fine until I tried to build the apk file. Some of packages (pyyaml, for instance) presented the same error during the process, but I solved it with pip install <package> or pip3 install <package>. But I could not find a way to fix the following packages: polars, scipy, torch, torchvision, psutil and ultralytics-thop. They are installed, by the way. Any idea? It is my first project using Utralytics. 

Bellow you have the result for polars, but it happens to all of the packages I have mentioned above.

(proj_flet) PS D:\desktop_21-01-2019\desktop\proj_flet\python_flet\> flet build apk -vv
[16:53:04] Run subprocess: ['C:\\flutter\\bin\\flutter.BAT', '--version', '--no-version-check', '--suppress-analytics']
[16:53:10] 1 Failed to validate Flutter version.
           Flutter 3.29.2 installed ✅                                                                                                                                                      
           Flutter executable: C:\Users\Dani\flutter\3.29.2\bin\flutter.bat
           Dart executable: C:\Users\Dani\flutter\3.29.2\bin\dart.bat
           JDK installed at C:\Users\Dani\java\17.0.13+11
           Configuring Flutter's path to JDK
           Run subprocess: ['C:\\Users\\Dani\\flutter\\3.29.2\\bin\\flutter.bat', 'config', '--no-version-check', '--suppress-analytics',
           '--jdk-dir=C:\\Users\\Dani\\java\\17.0.13+11']
[16:53:13] Setting "jdk-dir" value to "C:\Users\Dani\java\17.0.13+11".
           
           You may need to restart any open editors for them to read new settings.
           JDK installed ✅                                                                                                                                                                 
           Android SDK installation found at C:\Users\Dani\AppData\Local\Android\Sdk
           Android SDK package "cmdline-tools;latest" is already installed
           Android SDK package "platform-tools" is already installed
           Android SDK package "platforms;android-35" is already installed
           Android SDK package "build-tools;34.0.0" is already installed
           Android SDK installed ✅                                                                                                                                                         
[16:53:21] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref "0.28.3" ✅                                                                                      
           Run subprocess: ['C:\\Users\\Dani\\flutter\\3.29.2\\bin\\dart.bat', 'run', '--suppress-analytics', 'serious_python:main', 'package',
           'D:\\desktop_21-01-2019\\desktop\\proj_flet\\python_flet\\src', '--platform', 'Android', '-r', 'certifi==2025.8.3', '-r', 'flet==0.28.3', '-r',   
           'matplotlib>=3.3.0', '-r', 'numpy>=1.23.0', '-r', 'opencv-python>=4.6.0', '-r', 'pillow>=7.1.2', '-r', 'polars', '-r', 'psutil', '-r', 'pyyaml>=5.3.1', '-r',
           'requests>=2.23.0', '-r', 'scipy>=1.4.1', '-r', 'torch>=1.8.0', '-r', "torch>=1.8.0,!=2.4.0; sys_platform == 'win32'", '-r', 'torchvision>=0.9.0', '-r',
           'ultralytics-thop>=2.0.0', '--exclude', 'build', '--cleanup-packages', '--verbose']
[16:53:53] ERROR: Could not find a version that satisfies the requirement polars (from versions: none)
           ERROR: No matching distribution found for polars
           <_io.TextIOWrapper name=4 encoding='utf-8'>
           Run subprocess: ['C:\\Users\\Dani\\flutter\\3.29.2\\bin\\flutter.bat', 'doctor', '--no-version-check', '--suppress-analytics']
[16:54:31] Doctor summary (to see all details, run flutter doctor -v):
           [√] Flutter (Channel stable, 3.29.2, on Microsoft Windows , locale pt-BR)
           [√] Windows Version (10 Pro 64 bits, 2004)
           [√] Android toolchain - develop for Android devices (Android SDK version 36.0.0)
           [√] Chrome - develop for the web
           [√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.14.15 (September 2025))
           [√] Android Studio (version 2023.2)
           [√] VS Code (version 1.104.1)
           [√] Connected device (3 available)
           [√] Network resources

           • No issues found!

My dependencies in pyproject.toml:

dependencies = [
    "flet==0.28.3",
    "certifi==2025.8.3",
    "numpy>=1.23.0",
    "opencv-python>=4.6.0",
    "pillow>=7.1.2",
    "requests>=2.23.0",
    "matplotlib>=3.3.0",
    "pyyaml>=5.3.1",
    "scipy>=1.4.1",
    "torch>=1.8.0",
    "torch>=1.8.0,!=2.4.0; sys_platform == 'win32'",
    "torchvision>=0.9.0",
    "psutil",
    "polars",
    "ultralytics-thop>=2.0.0",
]

I do not have a compatible GPU, so I used colab to generate the .tflite file. Is that a problem?
So in my main.py I have:

from ultralytics import YOLO
model = YOLO('storage/data/yolo11n_float32.tflite')

Installing Ultralytics:

(proj_flet) PS D:\desktop_21-01-2019\desktop\proj_flet\python_flet> pip install ultralytics
Requirement already satisfied: ultralytics in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (8.3.203)
Requirement already satisfied: numpy>=1.23.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (2.2.6)
Requirement already satisfied: matplotlib>=3.3.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (3.10.3)
Requirement already satisfied: opencv-python>=4.6.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (4.10.0.84)
Requirement already satisfied: pillow>=7.1.2 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (11.2.1)
Requirement already satisfied: pyyaml>=5.3.1 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (5.3.1)
Requirement already satisfied: requests>=2.23.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (2.23.0)
Requirement already satisfied: scipy>=1.4.1 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (1.15.3)
Requirement already satisfied: torch>=1.8.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (2.7.1)
Requirement already satisfied: torchvision>=0.9.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (0.22.1)
Requirement already satisfied: psutil in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (7.0.0)
Requirement already satisfied: polars in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (1.33.1)
Requirement already satisfied: ultralytics-thop>=2.0.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from ultralytics) (2.0.0)
Requirement already satisfied: contourpy>=1.0.1 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from matplotlib>=3.3.0->ultralytics) (1.3.2) 
Requirement already satisfied: cycler>=0.10 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from matplotlib>=3.3.0->ultralytics) (0.12.1)    
Requirement already satisfied: fonttools>=4.22.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from matplotlib>=3.3.0->ultralytics) (4.58.2)
Requirement already satisfied: kiwisolver>=1.3.1 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from matplotlib>=3.3.0->ultralytics) (1.4.8)
Requirement already satisfied: packaging>=20.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from matplotlib>=3.3.0->ultralytics) (23.2)   
Requirement already satisfied: pyparsing>=2.3.1 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from matplotlib>=3.3.0->ultralytics) (3.2.3) 
Requirement already satisfied: python-dateutil>=2.7 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from matplotlib>=3.3.0->ultralytics) (2.9.0.post0)
Requirement already satisfied: six>=1.5 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from python-dateutil>=2.7->matplotlib>=3.3.0->ultralytics) (1.17.0)
Requirement already satisfied: chardet<4,>=3.0.2 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from requests>=2.23.0->ultralytics) (3.0.4)
Requirement already satisfied: idna<3,>=2.5 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from requests>=2.23.0->ultralytics) (2.10)       
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from requests>=2.23.0->ultralytics) (1.25.11)
Requirement already satisfied: certifi>=2017.4.17 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from requests>=2.23.0->ultralytics) (2025.8.3)
Requirement already satisfied: filelock in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from torch>=1.8.0->ultralytics) (3.18.0)
Requirement already satisfied: typing-extensions>=4.10.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from torch>=1.8.0->ultralytics) (4.14.0)
Requirement already satisfied: sympy>=1.13.3 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from torch>=1.8.0->ultralytics) (1.14.0)        
Requirement already satisfied: networkx in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from torch>=1.8.0->ultralytics) (3.4.2)
Requirement already satisfied: jinja2 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from torch>=1.8.0->ultralytics) (3.1.6)
Requirement already satisfied: fsspec in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from torch>=1.8.0->ultralytics) (2025.5.1)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from sympy>=1.13.3->torch>=1.8.0->ultralytics) (1.3.0)
Requirement already satisfied: MarkupSafe>=2.0 in d:\desktop_21-01-2019\desktop\proj_flet\lib\site-packages (from jinja2->torch>=1.8.0->ultralytics) (3.0.2)

Thanks,

**Comment by UltralyticsAssistant** [comment]
👋 Hello @danielefernandesdesigner, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will review your issue and assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us investigate your Flet APK build issue, please share:
-Exact flet build command used (with flags) and full -vv output as text
-Your pyproject.toml (dependencies and tool configs) and any build constraints files
-Python version, OS/arch, and whether you’re targeting CPU-only
-A minimal sample app that imports ultralytics and loads your YOLO11 TFLite model
-Confirmation that the same dependencies install and import successfully in a clean Python environment outside Flet

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
I doubt you can build an app to use Ultralytics just like that. All those dependencies are OS specific. You can't just use them on Android, nor are they designed for Android.

If you want to use YOLO in an Android App, you can check the Ultralytics Flutter plugin:

https://github.com/ultralytics/yolo-flutter-app

**Comment by DaniFernandesDesigner** [comment]
Hi **@Y-T-G**,  
Thank you for responding so quickly and for bringing back the Ultralytics Flutter plugin. I found it a few days ago, but unfortunately, I need to use Python to develop the application and have not been able to adapt the plugin to Flet. But maybe I gave up on the plugin too soon. After all, Flet is based on Flutter. I will keep trying. If I find a way to use it with Flet, I will post it here. If anyone has found a solution, it would be great to hear. If anyone can confirm that it's truly impossible to combine Flet with this plugin, that would also help me (and other people) save time. Thank you again.

**Comment by glenn-jocher** [comment]
Thanks, Dani—Android APKs can’t bundle the Ultralytics Python stack; packages like `torch`, `torchvision`, `scipy`, `polars` and `ultralytics-thop` don’t provide Android wheels, so `flet build apk` will fail. The supported path is to keep Python/Ultralytics off-device, export YOLO11 to TFLite, and run it on Android via the platform TFLite runtime (Flutter/Flet plugin), not through `YOLO()` in-app; your Colab-exported `.tflite` is fine, and re-exporting with NMS can simplify postprocessing:

```python
from ultralytics import YOLO
YOLO("yolo11n.pt").export(format="tflite", nms=True)
```

See the YOLO11 TFLite export guide for steps and options and the Android notes on delegates/quantization for performance considerations; whether Flet can wire a Flutter TFLite plugin is a Flet question we can’t confirm here, but that’s the right direction.  

- [YOLO11 TFLite export guide](https://docs.ultralytics.com/integrations/tflite/)  
- [Android runtime and acceleration notes](https://docs.ultralytics.com/hub/app/android/)

**Comment by DaniFernandesDesigner** [comment]
Hello, Glenn! First of all, thanks for Ultralytics. It's truly amazing! And thanks for saving me time by pointing me in the right direction and for bringing NMS to my attention. I will be using an old laptop to run my code; for now, that will do it. And in a few months, I will try to ditch Python and just use Flutter. Thanks again. I'm closing this issue.

**Comment by DaniFernandesDesigner** [comment]
Closing the issue.

**Comment by glenn-jocher** [comment]
Thanks, Dani—glad that helped; credit goes to the YOLO community and the Ultralytics team. Running on the laptop is fine; when you switch to Flutter, export an INT8 TFLite with NMS to keep mobile inference fast and integration simple:

```python
from ultralytics import YOLO
YOLO("yolo11n.pt").export(format="tflite", int8=True, nms=True, data="path/to/your.yaml", fraction=0.2)
```

If anything new comes up, please open a fresh issue; export args are detailed in the [exporter reference (TFLite/INT8 options)](https://docs.ultralytics.com/reference/engine/exporter/).
